src/files/glob-source.js

```diff
@@ -77,15 +107,29 @@ async function * toGlobSource ({ path, type, prefix }, options) {
 
   const globOptions = Object.assign({}, options.glob, {
     cwd: path,
-    nodir: true,
+    nodir: false,
     realpath: false,
     absolute: true
   })
 
   for await (const p of glob(path, '**/*', globOptions)) {
+    if (preserveMode || preserveMtime) {
+      const stat = await fs.stat(p)
```

Comment:
In the future we might get some perf wins by [parallel mapping](https://www.npmjs.com/package/streaming-iterables#parallelmap) these.

Reply:
Possibly - our bottlenecks typically aren't from reading paths from the filesystem but it's worth measuring before jumping to conclusions.  Like I just did.  :trollface: